WordPress/gutenberg
# [Create-block] Add `--variant` flag to allow creation of different block type variants

## What?
This PR introduces a new CLI flag to the `create-block` tool to indicate if the generated block should be dynamic or static. The default state is to create a static block and by passing `--variant dynamic` a dynamic block would be created.

## Why?
There is currently no way to scaffold a dynamic block using `create-block` without using a custom template. The intent here is to allow any template, including the built-in ones, to support creating either type of block.

## How?
I have added the new flag, `--variant`, and am currently using a combination of Mustache template conditionals and custom-named templates to load the correct files.

## Testing Instructions
1. Run `node index.js my-test --variant dynamic` from the root of the create-block package
2. Ensure that there is a `template.php` file added to the src directory and that no `save.js` file was added or is referenced in `index.js`
3. Run `node index.js my-test ` from the root of the create-block package and confirm that `save.js` is in-place and references correctly in `index.js`
